src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2393,6 +2393,27 @@ update_signature (MonoMethod *accessor_method)
 	g_assert_not_reached ();
 }
 
+static MonoClass *
+get_container_class (MonoClass *klass)
+{
+	return mono_class_is_ginst (klass) ? mono_class_get_generic_class (klass)->container_class : klass;
+}
```

Comment:
why?  this already exists: `mono_class_get_generic_type_definition` in `class-internals.h` does this

Reply:
Changed already.